Network Policy Reconciler - Update Buffer Fails

### Description
When i.e. deleting a network policy that is not managed by Karydia, the logs will print out an error message:
```
{"component":"networkpolicy_reconciler","level":"info","msg":"[INFO] Start network policy reconciler (syncNetworkPolicyHandler) for 'test-np' in namespace 'default'","time":"2019-12-18T16:16:06Z"}
{"component":"networkpolicy_reconciler","level":"error","msg":"[ERROR] Failed to get karydia network policy 'test-np'","time":"2019-12-18T16:16:06Z"}
{"component":"networkpolicy_reconciler","level":"warning","msg":"[WARN] Failed to get default network policy 'test-np'","time":"2019-12-18T16:16:06Z"}
{"component":"networkpolicy_reconciler","level":"info","msg":"[INFO] Successfully reconciled network policy 'karydia-default-network-policy' in namespace 'default'","time":"2019-12-18T16:16:06Z"}
{"component":"networkpolicy_reconciler","level":"info","msg":"[INFO] Successfully synced network policy 'default/test-np'","time":"2019-12-18T16:16:06Z"}
```
This error message states that the self-managed network policy cannot be found by Karydia. However, Karydia should not handle this network policy, as it is no default network policy of Karydia.


### Steps to reproduce
1. Create a new network policy
```
cat <<EOF | kubectl apply -f -
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: test-np
spec:
  podSelector:
    matchLabels: {}
EOF
```
2. Delete the newly created network policy
```
kubectl delete networkpolicy test-np
```
3. Check the logs of Karydia
```
kubectl logs -n karydia $(kubectl get pods -n karydia -l app=karydia -o jsonpath='{.items[0].metadata.name}') -f -c karydia
```

### Expected behavior
Karydia should ignore network policy that are not managed as a karydia-default-network-policy. Thus, there should be no error, when a networkpolicy is deleted that is not managed by Karydia. A correct log should look like this:
```
{"component":"networkpolicy_reconciler","level":"info","msg":"[INFO] Start network policy reconciler (syncNetworkPolicyHandler) for 'test-np' in namespace 'default'","time":"2019-12-18T16:40:43Z"}
{"component":"networkpolicy_reconciler","level":"info","msg":"[INFO] Successfully reconciled network policy 'karydia-default-network-policy' in namespace 'default'","time":"2019-12-18T16:40:43Z"}
```

--- a/pkg/controller/networkpolicy_reconciler.go
+++ b/pkg/controller/networkpolicy_reconciler.go
@@ -271,7 +271,7 @@ func (reconciler *NetworkpolicyReconciler) syncNetworkPolicy(namespaceName strin
 		}
 	}
 
-	if npName == reconciler.defaultNetworkPolicies[npName].GetName() && stringInSlice(npName, expectedNpNames) {
+	if stringInSlice(npName, expectedNpNames) && npName == reconciler.defaultNetworkPolicies[npName].GetName() {
 		networkPolicy, err := reconciler.networkPolicyLister.NetworkPolicies(namespaceName).Get(npName)
 		if err != nil {
 			if errors.IsNotFound(err) {

--- a/pkg/controller/networkpolicy_reconciler.go
+++ b/pkg/controller/networkpolicy_reconciler.go
@@ -260,7 +260,7 @@ func (reconciler *NetworkpolicyReconciler) syncNetworkPolicyHandler(key string) 
 }
 
 func (reconciler *NetworkpolicyReconciler) syncNetworkPolicy(namespaceName string, npName string, setting Setting) error {
-        expectedNpNames := strings.Split(setting.value, defaultNetworkPoiliciesDelimiter)
+	expectedNpNames := strings.Split(setting.value, defaultNetworkPoiliciesDelimiter)
 
 	for _, expectedNpName := range expectedNpNames {
 		if _, ok := reconciler.defaultNetworkPolicies[expectedNpName]; !ok {
@@ -269,6 +269,7 @@ func (reconciler *NetworkpolicyReconciler) syncNetworkPolicy(namespaceName strin
 				return nil
 			}
 		}
+	}
 
 	if npName == reconciler.defaultNetworkPolicies[npName].GetName() && stringInSlice(npName, expectedNpNames) {
 		networkPolicy, err := reconciler.networkPolicyLister.NetworkPolicies(namespaceName).Get(npName)

--- a/pkg/controller/networkpolicy_reconciler.go
+++ b/pkg/controller/networkpolicy_reconciler.go
@@ -180,7 +180,6 @@ func (reconciler *NetworkpolicyReconciler) processNextNetworkPolicyWorkItem() bo
 		}
 
 		reconciler.networkPolicyworkqueue.Forget(obj)
-		reconciler.log.Infof("Successfully synced network policy '%s'", key)
 		return nil
 	}(obj)
 
@@ -217,7 +216,6 @@ func (reconciler *NetworkpolicyReconciler) processNextNamespaceWorkItem() bool {
 		}
 
 		reconciler.namespaceWorkqueue.Forget(obj)
-		reconciler.log.Infof("Successfully synced workitem '%s'", key)
 		return nil
 	}(obj)
 

--- a/pkg/controller/networkpolicy_reconciler.go
+++ b/pkg/controller/networkpolicy_reconciler.go
@@ -260,14 +260,15 @@ func (reconciler *NetworkpolicyReconciler) syncNetworkPolicyHandler(key string) 
 }
 
 func (reconciler *NetworkpolicyReconciler) syncNetworkPolicy(namespaceName string, npName string, setting Setting) error {
-	if _, ok := reconciler.defaultNetworkPolicies[npName]; !ok {
-		if err := reconciler.updateBuffer(npName); err != nil {
-			reconciler.log.Warnf("Failed to get default network policy '%s'", npName)
-			return nil
-		}
-	}
+        expectedNpNames := strings.Split(setting.value, defaultNetworkPoiliciesDelimiter)
 
-	expectedNpNames := strings.Split(setting.value, defaultNetworkPoiliciesDelimiter)
+	for _, expectedNpName := range expectedNpNames {
+		if _, ok := reconciler.defaultNetworkPolicies[expectedNpName]; !ok {
+			if err := reconciler.updateBuffer(expectedNpName); err != nil {
+				reconciler.log.Warnf("Failed to get default network policy '%s'", expectedNpName)
+				return nil
+			}
+		}
 
 	if npName == reconciler.defaultNetworkPolicies[npName].GetName() && stringInSlice(npName, expectedNpNames) {
 		networkPolicy, err := reconciler.networkPolicyLister.NetworkPolicies(namespaceName).Get(npName)

